runtime/Java/src/org/antlr/v4/runtime/UnbufferedCharStream.java

```diff
@@ -82,7 +82,7 @@ public UnbufferedCharStream() {
 	/** Useful for subclasses that pull char from other than this.input. */
 	public UnbufferedCharStream(int bufferSize) {
 		n = 0;
-		data = new char[bufferSize];
+		data = new int[bufferSize];
 	}
 
 	public UnbufferedCharStream(InputStream input) {
```

Comment:
ok, great! could you please make the changes?  Does that mean we change this field to InputStream?

```
    protected Reader input;
```